agents/openai_functions_agent.go

```diff
@@ -0,0 +1,212 @@
+package agents
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/tmc/langchaingo/callbacks"
+	"github.com/tmc/langchaingo/llms"
+	"github.com/tmc/langchaingo/prompts"
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/tools"
+)
+
+// OpenAIFunctionsAgent is an Agent driven by OpenAIs function powered API.
+type OpenAIFunctionsAgent struct {
+	// LLM is the llm used to call with the values. The llm should have an
+	// input called "agent_scratchpad" for the agent to put its thoughts in.
+	LLM    llms.LanguageModel
+	Prompt prompts.FormatPrompter
+	// Chain chains.Chain
+	// Tools is a list of the tools the agent can use.
+	Tools []tools.Tool
+	// Output key is the key where the final output is placed.
+	OutputKey string
+	// CallbacksHandler is the handler for callbacks.
+	CallbacksHandler callbacks.Handler
+}
+
+var _ Agent = (*OpenAIFunctionsAgent)(nil)
+
+/*
```

Comment:
I don't think this comment is attached to anything. please attach it to NewOpenAIFunctionsAgent and start it with "NewOpenAIFunctionsAgent .." -- go documentation doesn't follow the layout you have here and the your arg names don't line up with the arguments.

Reply:
I got it.